jest/__mocks__/crypto.js

```diff
@@ -0,0 +1,31 @@
+//subtle object
+export const mock_crypto_subtle_digest = jest.fn();
+export const mock_crypto_subtle_importKey = jest.fn();
+export const mock_crypto_subtle_encrypt = jest.fn();
+export const mock_crypto_subtle_decrypt = jest.fn();
+export const mock_crypto_subtle_verify = jest.fn();
+
+const Subtle = jest.fn().mockImplementation(() => {
+    return {
+        digest: mock_crypto_subtle_digest,
+        importKey: mock_crypto_subtle_importKey,
+        encrypt: mock_crypto_subtle_encrypt,
+        decrypt: mock_crypto_subtle_decrypt,
+        verify: mock_crypto_subtle_verify
+    };
+});
+
+//crypto object
+export const mock_crypto_getRandomValues = jest.fn();
+
+const Crypto = jest.fn().mockImplementation(() => {
+    return {
+        getRandomValues: mock_crypto_getRandomValues,
+        subtle: new Subtle()
+    }
+});
+export const crypto = new Crypto();
+
+//are not property of either crypto or subtle
+export const mock_pem2ab = jest.fn();
+export const pem2ab = mock_pem2ab;
```

Comment:
This can be simplified with  `export const pem2ab = jest.fn();`?

Reply:
I chose keeping both `mock_pem2ab` and `pem2ab()` because:
1. `In `the main.test.js`, I preferred using `mock_pem2ab` instead of `pem2ab` `(expect(mock_pem2ab).toHaveBeenCalledTimes(1);)`  for consistency + to make it clear that we are using mock function. 
2. But in the `main.js`, I preferred using `pem2ab()` just like our customer will call.

Reply:
Ok, that makes sense